webapp/tests/karma/ts/modules/messages/messages.component.spec.ts

```diff
@@ -172,13 +201,128 @@ describe('Messages Component', () => {
       expect(component.loading).to.be.false;
       expect(component.conversations).to.eql( newConversations);
     });
+
+    it('ngOnDestroy() should unsubscribe from observables', () => {
+      const spySubscriptionsUnsubscribe = sinon.spy(component.subscriptions, 'unsubscribe');
+
+      component.ngOnDestroy();
+
+      expect(spySubscriptionsUnsubscribe.callCount).to.equal(1);
+    });
   });
 
-  it('ngOnDestroy() should unsubscribe from observables', () => {
-    const spySubscriptionsUnsubscribe = sinon.spy(component.subscriptions, 'unsubscribe');
+  describe('Messages breadcrumbs', () => {
+    const offlineUserContactDoc = {
+      _id: 'user',
+      parent: {
+        _id: 'parent',
+        name: 'parent',
+        parent: userContactGrandparent,
+      },
+    };
+    const offlineUserContactDoc2 = {
+      _id: 'user',
+      parent: {
+        _id: 'parent',
+        name: 'CHW Bettys Area',
+        parent: userContactGrandparent,
+      },
+    };
+    const conversations =  [
+      { key: 'a',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village', 'CHW Bettys Area']
+      },
+      { key: 'b',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village']
+      },
+      { key: 'c',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village', 'Ramdom Place']
+      },
+      { key: 'd',
+        message: { inAllMessages: true },
+        lineage : []
+      },
+      { key: 'e',
+        message: { inAllMessages: true },
+      },
+    ];
+
+    it('it should retrieve the hierarchy level of the connected user', () => {
+      expect(component.currentLevel).to.equal('parent');
+    });
+
+    it('should not alter conversations when user is offline and parent place isnt relevant to the conversation',
+      fakeAsync( () => {
+        sinon.resetHistory();
+
+        messageContactService.getList.resolves(conversations);
+        userContactService.get.resolves(offlineUserContactDoc);
+        sessionService.isOnlineOnly.returns(false);
+
+        component.updateConversations({merge : true});
+        tick();
+
+        expect(messageContactService.getList.callCount).to.equal(1);
+        expect(component.currentLevel).to.equal('parent');
+        expect(component.conversations).to.deep.equal(conversations);
+      }));
+
+    it('should not change the conversations lineage if the connected user is online only', fakeAsync( () => {
+      sinon.resetHistory();
+
+      messageContactService.getList.resolves(conversations);
+      userContactService.get.resolves(userContactDoc);
```

Comment:
Yeah we can use that scenario as well but with `isOnlineOnly.returns(true)`  it's not  necessary to call `userContactService.get.resolves(userContactDoc)` and  `expect(component.currentLevel).to.equal('CHW Bettys Area')` because it's irrelevant as `component.currentLevel` is only set when user is offline. 
The current value of `component.currentLevel ` only comes from the constructor in ngOnInit and not from `userContactService.get.resolves(userContactDoc)` consequences

Reply:
Yeah we can use that scenario as well but with `isOnlineOnly.returns(true)`  it's not  necessary to call `userContactService.get.resolves(userContactDoc)` and  `expect(component.currentLevel).to.equal('CHW Bettys Area')` because it's irrelevant as `component.currentLevel` is only set when user is offline. 
The current value of `component.currentLevel ` only comes from the constructor in ngOnInit and not from `userContactService.get.resolves(userContactDoc)` consequences

Reply:
so if you are confortable, we can do something like this :

```
    it('should not change the conversations lineage if the connected user is online only', fakeAsync( () => {
      sinon.resetHistory();

      messageContactService.getList.resolves(conversations);
      userContactService.get.resolves(bettyOfflineUserContactDoc);
      sessionService.isOnlineOnly.returns(true);

      component.ngOnInit();
      tick();
      component.updateConversations({merge : true});
      tick();

      expect(component.conversations).to.deep.equal(conversations);
    }));
```